src/snyk/cli/staticCliApi.ts

```diff
@@ -0,0 +1,84 @@
+import axios, { CancelTokenSource } from 'axios';
+import { IConfiguration } from '../common/configuration/configuration';
+import { PROTOCOL_VERSION } from '../common/constants/languageServer';
+import { DownloadAxiosResponse } from '../common/download/downloader';
+import { ILog } from '../common/logger/interfaces';
+import { getAxiosConfig } from '../common/proxy';
+import { IVSCodeWorkspace } from '../common/vscode/workspace';
+import { CliExecutable } from './cliExecutable';
+import { CliSupportedPlatform } from './supportedPlatforms';
+
+export interface IStaticCliApi {
+  getLatestCliVersion(releaseChannel: string): Promise<string>;
+  downloadBinary(platform: CliSupportedPlatform): Promise<[Promise<DownloadAxiosResponse>, CancelTokenSource]>;
+  getSha256Checksum(version: string, platform: CliSupportedPlatform): Promise<string>;
+}
+
+export class StaticCliApi implements IStaticCliApi {
+  constructor(
+    private readonly workspace: IVSCodeWorkspace,
+    private readonly configuration: IConfiguration,
+    private readonly logger: ILog,
+  ) {}
+
+  getLatestVersionDownloadUrl(releaseChannel: string): string {
+    const downloadUrl = `${this.configuration.getCliBaseDownloadUrl()}/cli/${releaseChannel}/ls-protocol-version-${PROTOCOL_VERSION}`;
+    return downloadUrl;
+  }
+
+  getDownloadUrl(version: string, platform: CliSupportedPlatform): string {
+    if (!version.startsWith('v')) {
+      version = `v${version}`;
+    }
+    const downloadUrl = `${this.configuration.getCliBaseDownloadUrl()}/cli/${version}/${this.getFileName(platform)}`;
+    return downloadUrl;
+  }
+
+  getSha256DownloadUrl(version: string, platform: CliSupportedPlatform): string {
+    const downloadUrl = `${this.getDownloadUrl(version, platform)}.sha256`;
+    return downloadUrl;
+  }
+
+  getFileName(platform: CliSupportedPlatform): string {
+    return CliExecutable.getFileName(platform);
+  }
+
+  async getLatestCliVersion(releaseChannel: string): Promise<string> {
+    let { data } = await axios.get<string>(
+      this.getLatestVersionDownloadUrl(releaseChannel),
+      await getAxiosConfig(this.workspace, this.configuration, this.logger),
```

Comment:
Historically, we sometimes had problems with Axios & proxies. Is this initialized correctly (like before)?